server/conn.go

```diff
@@ -1075,6 +1080,11 @@ func (cc *clientConn) Run(ctx context.Context) {
 							zap.Error(err),
 						)
 					}
+				} else if errors.ErrorEqual(err, errNetPacketTooLarge) {
+					err := cc.writeError(ctx, err)
```

Comment:
Some weird behavior I encountered:
```
tidb> set @@max_allowed_packet=0;
tidb> CREATE TABLE WALLET(user_id int, currency_id int, amount int, expiry_time DATETIME, is_lifetime int, pack_id int, sub_org_id int);
```
Then(see attached sql.txt):
```
tidb> source sql.txt
```
* MySQL:
```
mysql> source sql.txt
ERROR 2006 (HY000): MySQL server has gone away
No connection. Trying to reconnect...
Connection id:    15
Current database: *** NONE ***

ERROR 1153 (08S01): Got a packet bigger than 'max_allowed_packet' bytes
mysql> source sql.txt
ERROR 2006 (HY000): MySQL server has gone away
No connection. Trying to reconnect...
Connection id:    16
Current database: *** NONE ***

ERROR 1153 (08S01): Got a packet bigger than 'max_allowed_packet' bytes
```

This PR:
```
tidb> source sql.txt
ERROR 2006 (HY000): MySQL server has gone away
No connection. Trying to reconnect...
Connection id:    431
Current database: test

ERROR 2006 (HY000): MySQL server has gone away
No connection. Trying to reconnect...
Connection id:    433
Current database: test

ERROR 2006 (HY000): MySQL server has gone away
tidb> source sql.txt
ERROR 2006 (HY000): MySQL server has gone away
No connection. Trying to reconnect...
Connection id:    435
Current database: test

ERROR 2006 (HY000): MySQL server has gone away
No connection. Trying to reconnect...
Connection id:    437
Current database: test

ERROR 2006 (HY000): MySQL server has gone away
```
However, if I set a breakpoint right here, and 'continue' when the debugger paused here, the result of TiDB is exactly the same with MySQL.

Reply:
[sql.txt](https://github.com/pingcap/tidb/files/8465258/sql.txt)

Reply:
I see, it is related to the `auto-tls`[https://docs.pingcap.com/tidb/dev/tidb-configuration-file#auto-tls] configuration, if it is set to `true`, the error message is `ERROR 2006 (HY000): MySQL server has gone away`.